vmware-event-router/hack/create_k8s_config.sh

```diff
@@ -0,0 +1,193 @@
+echo -n "OpenFaaS Admin Password: "
+read -s OPENFAAS_PASSWORD
+if [[ -z "${OPENFAAS_PASSWORD}" ]]; then
+    cecho "Please start over. No input entered." $red
+    exit 1
+fi
+echo ""
+
+cecho "Please confirm the following settings are correct:\n" $cyan
+echo -e "\tVCENTER_SERVER=${VCENTER_SERVER}"
+echo -e "\tVCENTER_USERNAME=${VCENTER_USERNAME}"
+echo -e "\tVCENTER_PASSWORD=${VCENTER_PASSWORD}"
+echo -e "\tOPENFAAS_PASSWORD=${OPENFAAS_PASSWORD}"
+
+echo -e "${cyan}"
+read -p "Do you want to proceed with the VEBA K8s deployment [y]: " answer
+case $answer in
+    [Yy]* ) cecho "Starting Deployment ..." $green;;
+    [Nn]* ) echo "Exiting ..." $red; exit;;
+    * ) echo "Exiting ..." $red; exit;;
+esac
+
+
+cecho "Creating vmware namespace" $green
+echo -e "\tkubectl create namespace vmware\n"
+kubectl create namespace vmware
+
+cecho "Deploying OpenFaaS" $green
+echo -e "\tkubectl create -f faas-netes/namespaces.yml"
+echo -e "\tkubectl -n openfaas create secret generic basic-auth --from-literal=basic-auth-user=admin --from-literal=basic-auth-password=${OPENFAAS_PASSWORD}"
+echo -e "\tkubectl create -f faas-netes/yaml"
+kubectl create -f faas-netes/namespaces.yml
+kubectl -n openfaas create secret generic basic-auth --from-literal=basic-auth-user=admin --from-literal=basic-auth-password="${OPENFAAS_PASSWORD}"
+kubectl create -f faas-netes/yaml
+
+OPENFAAS_GATEWAY=$(kubectl -n openfaas describe pods $(kubectl -n openfaas get pods | grep "gateway-" | awk '{print $1}') | grep "^Node:" | awk -F "/" '{print $2}')
+while [ 1 ];
+do
+    cecho "Waiting for OpenFaaS to be ready ..." $green
+    OPENFAAS_GATEWAY=$(kubectl -n openfaas describe pods $(kubectl -n openfaas get pods | grep "gateway-" | awk '{print $1}') | grep "^Node:" | awk -F "/" '{print $2}')
+    if [ ! -z ${OPENFAAS_GATEWAY} ]; then
+        break
+    fi
+done
+
+cecho "Creating VEBA deployment files..." $green
+cat > event-router-config.json << EOF
+[{
+        "type": "stream",
+        "provider": "vmware_vcenter",
+        "address": "https://${VCENTER_SERVER}:443/sdk",
+        "auth": {
+            "method": "user_password",
+            "secret": {
+                "username": "${VCENTER_USERNAME}",
+                "password": "${VCENTER_PASSWORD}"
+            }
+        },
+        "options": {
+            "insecure": "true"
+        }
+    },
+    {
+        "type": "processor",
+        "provider": "openfaas",
+        "address": "http://gateway.openfaas:8080",
+        "auth": {
+            "method": "basic_auth",
+            "secret": {
+                "username": "admin",
+                "password": "${OPENFAAS_PASSWORD}"
+            }
+        },
+        "options": {
+            "async": "false"
+        }
+    },
+    {
+        "type": "metrics",
+        "provider": "internal",
+        "address": "0.0.0.0:8080",
+        "auth": {
+            "method": "none"
+        }
+    }
+]
+EOF
+
+cat > event-router-k8s.yaml << __EVENT_ROUTER_CONFIG
```

Comment:
This can be directly pulled from our Github repo (raw YAML) and passed to `kubectl create -f `. Since you don't modify this file there's no need for the redundancy here in my opinion.

Reply:
Done